webproject/src/main/java/com/google/sps/data/UserRoom.java

```diff
@@ -22,4 +22,34 @@ public String getRoomId() {
     public String getUserEmailRoom() {
         return userEmailRoom;
     }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String userEmail;
+        private String roomId;
+        private String userEmailRoom;
+
+        public Builder setUserEmail(String userEmail) {
+            this.userEmail = userEmail;
+            return this;
+        }
+
+        public Builder setRoomId(String roomId) {
+            this.roomId = roomId;
+            return this;
+        }
+
+        public Builder setUserEmailRoom() {
+            this.userEmailRoom = this.userEmail + "_" + this.roomId;
```

Comment:
i think this should be set by when build() is called?

Reply:
I have changed it to UserRoomProto now